Description: Copy the default /etx/xdg menu stuff if on does not exist on the first run.
Origin: vendor
Author: <peter@thecodergeek.com>
--- a/obmenu
+++ b/obmenu
@@ -475,6 +475,35 @@
 			self.menu_path = sys.argv[1]
 		elif len(sys.argv) == 1:
 			self.menu_path = os.getenv("HOME") + "/.config/openbox/menu.xml"
+			if not os.path.isfile("%s/.config/openbox/menu.xml" % (os.getenv("HOME"))):
+				if not os.path.isdir("%s/.config/" % (os.getenv("HOME"))):
+					try:
+						os.mkdir("%s/.config/" % (os.getenv("HOME")))
+					except:
+						print "Error: Can't create ~/.config directory!"
+						return
+
+				if not os.path.isdir("%s/.config/openbox" % (os.getenv("HOME"))):
+					try:
+						os.mkdir("%s/.config/openbox" % (os.getenv("HOME")))
+					except:
+						print "Error: Can't create ~/.config/openbox directory!"
+						return
+
+				if not os.path.isfile("/etc/xdg/openbox/menu.xml"):
+					print "Error: Couldn't find default menu file!"
+					return
+
+				try:
+					orig = open("/etc/xdg/openbox/menu.xml", "r")
+					dest = open("%s/.config/openbox/menu.xml" % (os.getenv("HOME")), "w")
+					dest.write(orig.read())
+					orig.close()
+					dest.close()
+				except:
+					print "Error: Couldn't create default menu file!"
+					return
+
 		else:
 			print "Error: Wrong number of arguments"
 			print "Usage: obmenu /path/to/menu.xml"
